src/core/types/api/user.ts

```diff
@@ -2,19 +2,28 @@ type BaseUserType = {
   ID: number;
   Username: string;
   IsAdmin: boolean;
-  RestrictedTags: Array<number>;
-  Avatar: {
-    RelativeFilepath: string;
-    ID: number;
-  };
+  RestrictedTags: number[];
+  Avatar: string;
 };
 
+type BasePutUserType = { [TKey in Exclude<keyof BaseUserType, 'ID'>]?: BaseUserType[TKey] | null | undefined; };
+
+type BasePostUserType = PutUserType & { Password?: string | null | undefined };
+
 export type CommunitySitesType = {
   AniDB: boolean;
   Trakt: boolean;
   Plex: boolean;
 };
 
-export type ApiUserType = BaseUserType & { CommunitySites: Array<string> };
+export type ApiUserType = BaseUserType & { CommunitySites: string[] };
```

Comment:
Do we really need so many identical types just named differently? as when I read the code I get the impression that they are all different in some way, or is it to safeguard against calling the wrong hook type of thing? But then doesn't typescript just convert from one type to another if fields are the same? @harshithmohan is it warranted you think to have so many superficial types?

Reply:
I don't think we need so many types....
If we make ID and Password optional in the `UserType`, it's more than enough, IMO